test/bin/stress_testing.sh

```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
+#
+# This script enable Stress conditions at OS level in local or remote host
+# limiting resources (latency, bandwidth, packet loss, memory, disk...)
+
+set -uo pipefail
+
+SCRIPT_NAME=$(basename "$0")
+WONDERSHAPER="/usr/local/bin/wondershaper"
+SPEEDTEST_CLI="/usr/local/bin/speedtest-cli"
+
+# run commands
+function run {
+  if [ "${SSH_PORT:-}" ]; then
+    SSH_PORT_PARAM+="-p ${SSH_PORT}"
+  fi
+
+  if [ "${SSH_PKEY:-}" ]; then
+    SSH_PKEY_PARAM="-i ${SSH_PKEY}"
+  fi
+
+  if [ "${SSH_USER:-}" ]; then
+    SSH_USER_PARAM="${SSH_USER}@"
+  fi
+
+  if [ "${SSH_HOSTNAME:-}" ]; then
+    # shellcheck disable=SC2086
+    ssh -o LogLevel=ERROR ${SSH_PORT_PARAM:-} ${SSH_PKEY_PARAM:-} ${SSH_USER_PARAM:-}${SSH_HOSTNAME} "$@"
+  else
+    "$@"
+  fi
+}
+
+# bandwidth condition
+function bandwidth_condition {
+  bandwidth_condition_ready
+  if [ "${ACTION}" == "enable" ]; then
+    enable_bandwidth
+  elif [ "${ACTION}" == "disable" ]; then
+    disable_bandwidth
+  elif [ "${ACTION}" == "get" ]; then
+    get_bandwidth
+  fi
+}
+
+function bandwidth_condition_ready { 
+  # tc command
+  check_command "tc -V"
+
+  # wondershaper
+  if ( ! sudo ${WONDERSHAPER} -v  &> /dev/null);  then
+    run sudo curl -Lo ${WONDERSHAPER} https://raw.githubusercontent.com/magnific0/wondershaper/master/wondershaper &> /dev/null
+    run sudo chmod +x ${WONDERSHAPER} &> /dev/null
+    check_command "sudo ${WONDERSHAPER} -v"
+  fi
+
+  # speedtest-cli
+  if ( ! ${SPEEDTEST_CLI} --version &> /dev/null);  then
+    run sudo curl -Lo ${SPEEDTEST_CLI} https://raw.githubusercontent.com/sivel/speedtest-cli/master/speedtest.py &> /dev/null
+    run sudo chmod +x ${SPEEDTEST_CLI} &> /dev/null
+    check_command "${SPEEDTEST_CLI} --version"
+  fi
```

Comment:
This code is not going to work on ostree systems because /usr/local/bin is read-only.
Should we put these utilitties in the home directory instead?

Reply:
No need to change the directory. There's a symlink from `/usr/local` to `/var/usrlocal` on ostree systems. I tested it and it works.

Reply:
Hah, nice. I did not know that.